### ISSUE TITLE
Extract name from torrent file (AB)

### ISSUE THREAD

**Issue by carcinogoy** [autobrr/autobrr#238]
Hi would it be possible to add the option to extract the name of the torrent file for matching instead of the regex matching groups.

The indexer animebytes has announcements that are sometimes hard to parse.
https://regexr.com/6ed9d
Group 1 almost always has stuff in it you don't want. 

I had a working example on an old build but the refactoring has left me confused where to put this
`ExtractName		    bool      'json:"extract_name"'`
to internal\domain\filter.go

and then 
```
if f.ExtractName  {
  // attempt to use torrent title
  if torrentFileRes == nil {
	  torrentFileRes, err = release.DownloadTorrentFile(nil)
	  log.Debug().Msgf("filter-service.find_and_check_filters: (%v) got torrent info: %+v", f.Name, torrentInfo)
  }
  // parse torrent metainfo
  torrentMetaInfo, err = torrentFileRes.MetaInfo.UnmarshalInfo()	
  release.TorrentName = torrentMetaInfo.Name
}
```
to internal\filter\service.go

**Comment by ludviglundgren** [comment]
I think it's probably better to use their api to get the name than to download all torrents to parse. I'll try and get started on this soon.

**Comment by einsern** [comment]
> I think it's probably better to use their api to get the name than to download all torrents to parse. 
Yes definitely. autodl-irssi downloads every torrent in question and it marks every torrent as snatched but not downloaded on some sites. Not to mention the extra load torrent sites.

**Comment by zze0s** [comment]
With all the recent changes and massaging of the torrentName for ab to make it easier to parse, I'm closing this since this isn't something that will be integrated.
